packages/wallet/src/redux/channel-state/reducer.ts

```diff
@@ -84,12 +63,166 @@ const initializingChannels: ReducerWithSideEffects<states.InitializingChannelSta
       ...state,
       // We have to temporarily store the private key under the address, since
       // we can't know the channel id until both participants know their addresses.
-      [address]: states.waitForChannel({ address, privateKey }),
+      [address]: { address, privateKey },
     },
     sideEffects: { messageOutbox: channelInitializationSuccess(wallet.address) },
   };
 };
 
+type CommitmentReceived = actions.OwnCommitmentReceived | actions.OpponentCommitmentReceived;
+const handleFirstCommmit = (
+  state: states.ChannelState,
+  action: CommitmentReceived,
+): StateWithSideEffects<states.ChannelState> => {
```

Comment:
This function is pretty long so it might be nice to split it into two, something like `initializeChannel` and `handleCommitment`?

Reply:
I think I will leave this for a future refactor. I'm going to be doing some work on the channel state reducer as part of the challenge work.